### DISCUSSION TITLE
ColumnChart with bordered bins

### DISCUSSION THREAD

**Discussion by ThomasCapelli** [patrykandpatrick/vico#969] [Questions] [answered]
### Question

While working with the library for compose(vico-compose-m3 2.0.0-beta.3), I managed to create bins (bars) with a required stroke. How can I have a border in every bar side but not bottom, where there is the bottomAxis?

### Vico version(s)

vico:compose-m3:2.0.0-beta.3

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @ThomasCapelli. `LineComponent.stroke` is applied along every edge, as this is how strokes work with Android's `Paint`. However, you can achieve the result you're looking for by subclassing `LineComponent` and using `LayeredComponent`. Below is a simple example that can be inlined. If needed, you can convert this to a standalone subclass that accepts a `LayeredComponent` and a thickness value.

```kt
object : LineComponent(fill = Fill.Transparent, thicknessDp = 16f) {
  val layeredComponent =
    LayeredComponent(
      rememberShapeComponent(Fill.Black),
      rememberShapeComponent(fill(Color.Red)),
      insets(start = 4.dp, top = 4.dp, end = 4.dp),
    )

  override fun draw(
    context: DrawingContext,
    left: Float,
    top: Float,
    right: Float,
    bottom: Float,
  ) {
    layeredComponent.draw(context, left, top, right, bottom)
  }
}
```
![image](https://github.com/user-attachments/assets/25eee9a2-0512-41a9-9081-ce909bd62c63)

**Reply by ThomasCapelli** [reply]
Thanks for your answer! Anyway, there is still a big problem caused by my front component having a gradient as fill. In this way it interpolates with the background (black for example) resulting in the wrong color

**Reply by Gowsky** [reply]
Could you clarify? Is the gradient semi-transparent?

**Reply by ThomasCapelli** [reply]
Yes, the rear component, in my case, must have a yellow fill, while the front one must have vertical gradient fill going from yellow with 100% alpha to yellow with 10% alpha(so semi-transparent).

**Reply by Gowsky** [reply, marked as answer]
Got it. Following my previous example, we can add a custom shape.
```kt
fun threeEdgeShape(thickness: Dp) = Shape { context, path, left, top, right, bottom ->
  with(path) {
    val offset = context.run { thickness.value.pixels / 2 }
    moveTo(left, bottom + offset)
    lineTo(left, top)
    lineTo(right, top)
    lineTo(right, bottom + offset)
  }
}
```
And then modify the custom `LineComponent` like so.
```kt
object : LineComponent(fill = Fill.Black, thicknessDp = 16f) {
  val layeredComponent =
    LayeredComponent(
      rememberShapeComponent(
        fill(
          ShaderProvider.verticalGradient(
            arrayOf(
              Color.Yellow,
              Color.Yellow.copy(alpha = .1f),
            ),
          ),
        ),
      ),
      rememberShapeComponent(
        fill = Fill.Transparent,
        shape = ThreeEdgeShape(4.dp),
        strokeFill = Fill.Black,
        strokeThickness = 4.dp,
      ),
    )

  override fun draw(
    context: DrawingContext,
    left: Float,
    top: Float,
    right: Float,
    bottom: Float,
  ) {
    layeredComponent.draw(context, left, top, right, bottom)
  }
}
```
![image](https://github.com/user-attachments/assets/061a23da-9afc-4ed4-9490-d9c668baa5b8)

**Reply by ThomasCapelli** [reply]
Thank you for your support and for the amazing work to help the community!

**Reply by Gowsky** [reply]
You're welcome!
